src/main/java/net/minecraftforge/client/loading/ForgeLoadingOverlay.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.minecraftforge.client.loading;
+
+import com.mojang.blaze3d.platform.GlStateManager;
+import com.mojang.blaze3d.systems.RenderSystem;
+import com.mojang.blaze3d.vertex.BufferBuilder;
+import com.mojang.blaze3d.vertex.BufferUploader;
+import com.mojang.blaze3d.vertex.BufferVertexConsumer;
+import com.mojang.blaze3d.vertex.DefaultVertexFormat;
+import com.mojang.blaze3d.vertex.Tesselator;
+import com.mojang.blaze3d.vertex.VertexFormat;
+import com.mojang.blaze3d.vertex.VertexSorting;
+import net.minecraft.Util;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.GuiGraphics;
+import net.minecraft.client.gui.screens.LoadingOverlay;
+import net.minecraft.client.renderer.GameRenderer;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.packs.resources.ReloadInstance;
+import net.minecraft.util.Mth;
+import net.minecraftforge.fml.StartupMessageManager;
+import net.minecraftforge.fml.earlydisplay.ColourScheme;
+import net.minecraftforge.fml.earlydisplay.DisplayWindow;
+import net.minecraftforge.fml.loading.progress.ProgressMeter;
+import org.jetbrains.annotations.NotNull;
+import org.joml.Matrix4f;
+
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+import static com.mojang.blaze3d.platform.GlConst.*;
+import static net.minecraft.util.Mth.clamp;
+import static org.lwjgl.opengl.GL30C.glViewport;
+import static org.lwjgl.opengl.GL30C.glTexParameterIi;
+
+public class ForgeLoadingOverlay extends LoadingOverlay {
```

Comment:
A bit of documentation as to why this exists compared to the original would be very helpful in the future.